designs/consolidation-policies.md

```diff
@@ -0,0 +1,72 @@
+
+The separation of the single-node and multi-node consolidation is arbitrary. The single-node consolidation finds the "least costly" eligible node to disrupt, and uses that as a solution, even if it's a poor solution. However, the least-costly eligible node to disrupt is not necessarily the most cost-savings node to disrupt, but the single-node consolidation stops after it has any solution.
+
+In contrast, the multi-node consolidation spends as much time as possible to find a solution. So while price thresholds could be implemented into each consolidation controller, it is simpler to consider merging their responsabilities into one shared controller based on multi-node consolidation. This combined consolidation controller could search as long as possible for the most effective consolidation outcome, from emptiness, multi-node or single-node replacements.
+
+#### Pros/Cons
+* 👍👍 No new configuration for the `NodePool` crd
+* ~ More complex implementation for consolidation, but easier to understand (only 1 controller for consolidation).
+* 👎 It's possible a one-size fits all price threshold will satisfy some customers and not others
+* 👎 High amount of engineering effort to implement.
+
+### Option 2: Refactor and Expand Consolidation Policies
+Another approach would be to not make significant changes to consolidation, but expose controls to allow enabling or disabling various consolidation controllers. Karpenter already presents the binary consolidation options. The `NodePool` resource has a disruption field that exposes the `consolidationPolicy` field. This field is an enum with two supported values: `WhenEmpty` and `WhenUnderutilized`.
+
+The proposal is to add two new consolidation policy enum values:
+
+* `WhenCheaper`
+* `WhenUnderutilizedOrCheaper`
+
+And change the semantics of the existing consolidation policy enum value:
+
+* `WhenUnderutilized`
+
+The semantics of each of the enum values matches their names, making it straightforward to explain to customers or users. `WhenUnderutilizedOrCheaper` becomes the new default. `WhenUnderutilizedOrCheaper` is the same as `WhenUnderutilized` previously. `WhenUnderutilized` semantics changes to only allow emptiness or multi-node consolidation to occur for a nodepool. `WhenCheaper` only allows emptiness or single-node consolidation to occur for a nodepool.
+
+#### Pros/Cons
+* 👍 Simple configuration for the `NodePool` crd
+* 👍 Simple implementation for the various consolidation controllers.
+* 👎 If Karpenter adds consolidation modes with new purposes, more `consolidationPolicy` enum values will be needed to make them toggleable.
+
+### Option 3: Add Consolidation Configuration to NodePools
```

Comment:
not sure if we can do this in a backwards compatible way that wouldn't require a v2 to the API.